components/Shipping/interfaces.rst

```diff
@@ -0,0 +1,210 @@
+
+.. note::
+    For more detailed information go to `Sylius API ShippingSubjectInterface`_.
+
+.. _Sylius API ShippingSubjectInterface: http://api.sylius.org/Sylius/Component/Shipping/Model/ShippingSubjectInterface.html
+
+
+Calculator interfaces
+---------------------
+
+CalculatorInterface
+~~~~~~~~~~~~~~~~~~~
+
+This interface provides basic methods for calculators. Every custom calculator should implement **CalculatorInterface** or extends
+class **Calculator**, which has a basic implementation of methods from this interface.
+
+.. note::
+    For more detailed information go to `Sylius API CalculatorInterface`_.
+
+.. _Sylius API CalculatorInterface: http://api.sylius.org/Sylius/Component/Shipping/Calculator/CalculatorInterface.html
+
+DelegatingCalculatorInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by any object, which will be responsible for delegating the calculation to a correct calculator instance.
+
+.. note::
+    For more detailed information go to `Sylius API DelegatingCalculatorInterface`_.
+
+.. _Sylius API DelegatingCalculatorInterface: http://api.sylius.org/Sylius/Component/Shipping/Calculator/DelegatingCalculatorInterface.html
+
+
+CalculatorRegistryInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by an object, which will keep all calculators registered inside container.
+
+.. note::
+    For more detailed information go to `Sylius API CalculatorRegistryInterface`_.
+
+.. _Sylius API CalculatorRegistryInterface: http://api.sylius.org/Sylius/Component/Shipping/Calculator/Registry/CalculatorRegistryInterface.html
+
+Checker Interfaces
+------------------
+
+.. _component_shipping_checker_registry_rule-checker-registry-interface:
+
+RuleCheckerRegistryInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by an service responsible for providing an information about available rule checkers.
+
+.. note::
+    For more detailed information go to `Sylius API RuleCheckerRegistryInterface`_.
+
+.. _Sylius API RuleCheckerRegistryInterface: http://api.sylius.org/Sylius/Component/Shipping/Checker/Registry/RuleCheckerRegistryInterface.html
+
+.. _component_shipping_checker_rule-checker-interface:
+
+RuleCheckerInterface
+~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by an object, which checks if shipping subject passes configured requirements.
+
+.. note::
+    For more detailed information go to `Sylius API RuleCheckerInterface`_.
+
+.. _Sylius API RuleCheckerInterface: http://api.sylius.org/Sylius/Component/Shipping/Checker/RuleCheckerInterface.html
+
+
+.. _component_shipping_checker_shipping-method-eligibility-checker-interface:
+
+ShippingMethodEligibilityCheckerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by an object, which checks if shipping method rules are capable of shipping given subject.
```

Comment:
` [...] which checks if the given shipping subject is eligible for the shipping method rules.`